src/backend.rs

```diff
@@ -173,9 +173,47 @@ pub enum LoadingStep {
         farm_index: u8,
         path: PathBuf,
     },
+    WipingFarmSuccessfully,
     WipingNode {
         path: PathBuf,
     },
+    WipingNodeSuccessfully,
+}
+
+impl LoadingStep {
+    fn basic_progress(&self) -> f32 {
+        match self {
+            LoadingStep::LoadingConfiguration => 0.0,
+            LoadingStep::ReadingConfiguration => 5.0,
+            LoadingStep::ConfigurationReadSuccessfully {
+                configuration_exists: _,
+            } => 10.0,
+            LoadingStep::CheckingConfiguration => 15.0,
+            LoadingStep::ConfigurationIsValid => 20.0,
+            LoadingStep::DecodingChainSpecification => 25.0,
+            LoadingStep::DecodedChainSpecificationSuccessfully => 30.0,
+            LoadingStep::CheckingNodePath => 35.0,
+            LoadingStep::CreatingNodePath => 40.0,
+            LoadingStep::NodePathReady => 45.0,
+            LoadingStep::PreparingNetworkingStack => 50.0,
+            LoadingStep::ReadingNetworkKeypair => 55.0,
+            LoadingStep::GeneratingNetworkKeypair => 60.0,
+            LoadingStep::WritingNetworkKeypair => 65.0,
+            LoadingStep::InstantiatingNetworkingStack => 70.0,
+            LoadingStep::NetworkingStackCreatedSuccessfully => 75.0,
+            LoadingStep::CreatingConsensusNode => 85.0,
+            LoadingStep::ConsensusNodeCreatedSuccessfully => 90.0,
+            LoadingStep::CreatingFarmer => 95.0,
+            LoadingStep::FarmerCreatedSuccessfully => 100.0,
+            LoadingStep::WipingFarm {
+                farm_index: _,
+                path: _,
+            } => 0.0,
+            LoadingStep::WipingFarmSuccessfully => 50.0,
+            LoadingStep::WipingNode { path: _ } => 80.0,
+            LoadingStep::WipingNodeSuccessfully => 100.0,
```

Comment:
Naively you'd think this would work, but it is far from accurate. Different steps take different amount of time and thus progress with fixed 5% between steps is not accurate.

Also it doesn't account for the fact that there could be multiple steps in node creation and it definitely doesn't account for how many farms there are. If you have one farm and 20 it will take vastly different amount of time to complete that step. I see that you take that into consideration below, but it is still 5% for all farms in total.

This is why https://github.com/subspace/space-acres/issues/133 says to do estimation first and scale everything accordingly to show realistic progress.